src/bacnet/bacapp.c

```diff
@@ -1721,7 +1691,6 @@ static int bacapp_decode_context_data_complex(
                     max_apdu_len - apdu_len, inner_tag_number, &tmpvalue, prop);
                 if (len >= 0) {
                     apdu_len += len;
-                    value = bacapp_allocate_new_value(&allocate_data, value, &tmpvalue);
```

Comment:
Probably should be renamed to bacapp_decode_context_data_complex_len or similar and the value param removed. Can be done later though.

Reply:
Sure will add a note to do that